mobile_packet_verifier/src/burner.rs

```diff
@@ -0,0 +1,287 @@
+            let escrow_account =
+                DelegatedDataCreditsV0::try_deserialize(&mut account_data)?.escrow_account;
+
+            tracing::info!("Burning {} DC from {}", amount, payer);
+
+            let instructions = {
+                let request = RequestBuilder::from(
+                    data_credits::id(),
+                    &self.cluster,
+                    std::rc::Rc::new(Keypair::from_bytes(&self.keypair).unwrap()),
+                    Some(CommitmentConfig::confirmed()),
+                    RequestNamespace::Global,
+                );
+
+                let accounts = accounts::BurnDelegatedDataCreditsV0 {
+                    sub_dao_epoch_info,
+                    dao: self.program_cache.dao,
+                    sub_dao: self.program_cache.sub_dao,
+                    account_payer: self.program_cache.account_payer,
+                    data_credits: self.program_cache.data_credits,
+                    delegated_data_credits: crate::pdas::delegated_data_credits(
+                        &self.program_cache.sub_dao,
+                        &payer,
+                    ),
+                    token_program: spl_token::id(),
+                    helium_sub_daos_program: helium_sub_daos::id(),
+                    system_program: solana_program::system_program::id(),
+                    dc_burn_authority: self.program_cache.dc_burn_authority,
+                    dc_mint: self.program_cache.dc_mint,
+                    escrow_account,
+                    registrar: self.program_cache.registrar,
+                };
+                let args = instruction::BurnDelegatedDataCreditsV0 {
+                    args: data_credits::BurnDelegatedDataCreditsArgsV0 { amount },
+                };
+
+                // As far as I can tell, the instructions function does not actually have any
+                // error paths.
+                request
+                    .accounts(accounts)
+                    .args(args)
+                    .instructions()
+                    .unwrap()
+            };
+
+            let blockhash = self.provider.get_latest_blockhash().await?;
+            let signer = Keypair::from_bytes(&self.keypair).unwrap();
+
+            let tx = SolanaTransaction::new_signed_with_payer(
+                &instructions,
+                Some(&signer.pubkey()),
+                &[&signer],
+                blockhash,
+            );
+
+            let signature = self.provider.send_and_confirm_transaction(&tx).await?;
+            tracing::info!(
+                "Successfully burned data credits. Transaction: {}",
+                signature
+            );
+
+            // Delete from the data transfer session and write out to S3
+
+            sqlx::query("DELETE FROM data_tranfer_sessions WHERE payer = $1")
+                .bind(payer)
+                .execute(&self.pool)
+                .await?;
+
+            for session in sessions {
+                self.valid_sessions
```

Comment:
auto_commit is set to false for this file_sink, but I don't see the call to commit anywhere.